lib/msal-core/src/cache/AuthCache.ts

```diff
@@ -153,11 +154,17 @@ export class AuthCache extends BrowserStorage {// Singleton
      */
     getAllAccessTokens(clientId: string, homeAccountIdentifier: string): Array<AccessTokenCacheItem> {
         const results = Object.keys(window[this.cacheLocation]).reduce((tokens, key) => {
-            if ( window[this.cacheLocation].hasOwnProperty(key) && key.match(clientId) && key.match(homeAccountIdentifier)) {
+            const keyMatches = key.match(clientId) && key.match(homeAccountIdentifier) && key.match(Constants.scopes);
+            if ( keyMatches ) {
                 const value = this.getItem(key);
                 if (value) {
-                    const newAccessTokenCacheItem = new AccessTokenCacheItem(JSON.parse(key), JSON.parse(value));
-                    return tokens.concat([ newAccessTokenCacheItem ]);
+                    try {
+                        const parseAtKey = JSON.parse(key);
+                        const newAccessTokenCacheItem = new AccessTokenCacheItem(parseAtKey, JSON.parse(value));
+                        return tokens.concat([ newAccessTokenCacheItem ]);
+                    } catch (e) {
+                        throw ClientAuthError.createCacheParseError(key);
```

Comment:
I would prefer we just pass the whole error also instead of the just the key since we are doing multiple operations within the 'try' block.

Reply:
see the idtokenparsing error in the same ClientAuthError.ts file.